services/motion/builtin/builtin_test.go

```diff
@@ -829,38 +878,106 @@ func TestCheckPlan(t *testing.T) {
 	err = newFS.AddFrame(cameraFrame, cameraOriginFrame)
 	test.That(t, err, test.ShouldBeNil)
 
-	t.Run("ensure transforms of obstacles works - no collision", func(t *testing.T) {
-		// create obstacle
-		obstacle, err := spatialmath.NewBox(
-			spatialmath.NewPoseFromPoint(r3.Vector{1500, -6, 0}),
-			r3.Vector{10, 10, 1}, "obstacle",
-		)
-		test.That(t, err, test.ShouldBeNil)
-		geoms := []spatialmath.Geometry{obstacle}
-		gifs := []*referenceframe.GeometriesInFrame{referenceframe.NewGeometriesInFrame(cameraFrame.Name(), geoms)}
+	type testCase struct {
```

Comment:
this now explicitly tests `CheckPlan` for differential drive bases only.
Because we want to only test this method against differential drive bases we no longer use `createMoveOnGlobeEnvironment`, `newMoveOnGlobeRequest` to create the testing env.
Instead, we construct a differential drive kinematic base, localizer, and frame system within the test itself.

The test cases themselves have also changed a bit.
The latter two tests `"non nil error state - ensure success"` and `"ensure transforms of obstacles works - collision with camera"` have been changed into `"non nil error state, partial plan, with no collision - ensure success"` and `"non nil error state, partial plan, with collision - ensure failure"` respectively.